pkg/hubble/metrics/cell/server.go

```diff
@@ -0,0 +1,145 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Hubble
+
+package metricscell
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"log/slog"
+	"net/http"
+	"sync"
+	"sync/atomic"
+
+	"github.com/cilium/hive/cell"
+
+	"github.com/cilium/cilium/api/v1/models"
+	"github.com/cilium/cilium/pkg/hubble/metrics"
+	"github.com/cilium/cilium/pkg/hubble/server/serveroption"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+)
+
+var (
+	statusWarnWaitCertsAvail = "Waiting for TLS certificates to become available"
+	statusErrWaitCertsAvail  = "Failed waiting for TLS certificates to become available"
+	statusErrServe           = "Failed serving HTTP metrics"
+)
+
+type Server interface {
+	Status() *models.HubbleMetricsStatus
+}
+
+type metricsServer struct {
+	logger *slog.Logger
+
+	server *http.Server
+
+	tlsConfigPromise tlsConfigPromise
+
+	wg        sync.WaitGroup
+	tlsCtx    context.Context
+	tlsCancel context.CancelFunc
+
+	statusWarn atomic.Pointer[string]
+	statusErr  atomic.Pointer[string]
+}
+
+func newMetricsServer(p params) Server {
+	if p.Config.MetricsServer == "" {
+		p.Logger.Info("The Hubble metrics server is disabled")
+		return nil
+	}
+
+	mux := metrics.ServerHandler(metrics.Registry, p.Config.EnableOpenMetrics)
+	ctx, cancel := context.WithCancel(context.Background())
+	server := &metricsServer{
+		logger: p.Logger,
+		server: &http.Server{
+			Addr:    p.Config.MetricsServer,
+			Handler: mux,
+		},
+		tlsConfigPromise: p.TLSConfigPromise,
+		tlsCtx:           ctx,
+		tlsCancel:        cancel,
```

Comment:
Can we use the hook context instead?

Reply:
See reply above. tldr: we can't, it doesn't have the appropriate lifetime.